.github/workflows/ci.yaml

```diff
@@ -0,0 +1,37 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: charmed-hpc-terraform tests
+on:
+  workflow_call:
+  pull_request:
+
+jobs:
+  inclusive-naming-check:
+    name: Inclusive naming check
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Run tests
+        uses: get-woke/woke-action@v0
+        with:
+          fail-on-error: true
+
+  validation-test:
+    name: Validation tests
+    runs-on: ubuntu-24.04
+    needs:
+      - inclusive-naming-check
```

Comment:
I don't think the `validation-test` job has a hard dependency on the `inclusive-naming-check` job completing successfully to run? 

Both are checking the style of the Charmed HPC TF plans their own way, so they can run concurrently to each other. Plus, it's better to give all the feedback on the new PR all at once rather than require the author to push a new commit to get feedback from `validation-test` as well.

Reply:
I think it's fair to require the simplest of checks to pass before running the chunky tests. This is what we also do on our charm repos, and it really works well to avoid wasting unnecessary CI time.

Reply:
I'd say it makes sense to have them run concurrently now and revisit as our tests become more complex in future. I've removed the dependency for the moment.

Reply:
@jedel1043 I definitely agree that we shouldn't try to waste CI time with chunky tests. _Looking at the Slurm charm integration tests take..._ :eyes: 

However, I see `opentofu validate` up there with `ruff check` where it's inexpensive and quick to run; it just checks to ensure that we don't have any "obvious" issues in the TF plans. If we added a bigger E2E test, like deploying a test cluster on Azure in CI (`just test azure`), then I would agree that we definitely need to gate that test on the less expensive tests.

Overall, I'd eventually like to see us follow the advice of the lightning talk we saw in The Hague where we add `pre-commit` to all our repos and have most of the inexpensive tests run before changes are even committed.